apps/gitness/src/components/FileEditor.tsx

```diff
@@ -153,158 +153,156 @@ export const FileEditor: React.FC = () => {
-                </ToggleGroupItem>
-              </ToggleGroup>
-            </StackedList.Item>
-          </StackedList.Root>
-          {view === 'contents' ? (
-            <CodeEditor
-              language={language}
-              codeRevision={{ code: content }}
-              onCodeRevisionChange={value => setContent(value?.code || '')}
-              themeConfig={themeConfig}
-              options={{
-                readOnly: false
+      <SandboxLayout.Content>
+        <ListActions.Root>
+          <ListActions.Left>
+            <ButtonGroup.Root spacing="2">
+              <BreadcrumbItem>
+                <BreadcrumbLink asChild>
+                  <Link to={repoPath}>
+                    <Text size={2} color="tertiaryBackground" className="hover:text-foreground">
+                      {repoId}
+                    </Text>
+                  </Link>
+                </BreadcrumbLink>
+              </BreadcrumbItem>
+              <Text size={2} color="tertiaryBackground" className="pt-2">
+                /
+              </Text>
+              {pathParts?.map((path: PathParts, index: number) => {
+                return (
+                  <>
+                    <BreadcrumbItem>
+                      <BreadcrumbLink asChild>
+                        <Link to={repoPath + '/~/' + path.parentPath}>
+                          <Text
+                            size={2}
+                            color="tertiaryBackground"
+                            className={cn('hover:text-foreground', {
+                              'text-primary': index === pathParts?.length - 1
+                            })}
+                          >
+                            {path.path}
+                          </Text>
+                        </Link>
+                      </BreadcrumbLink>
+                    </BreadcrumbItem>
+                    <Text size={2} color="tertiaryBackground" className="pt-2">
+                      /
+                    </Text>
+                  </>
+                )
+              })}
+              <BreadcrumbItem>
+                <Input
+                  id="fileName"
+                  value={fileName}
+                  size={20}
+                  onInput={(event: ChangeEvent<HTMLInputElement>) => {
+                    setFileName(event.currentTarget.value)
+                  }}
+                  ref={_ref => (inputRef.current = _ref)}
+                  onBlur={rebuildPaths}
+                  onFocus={({ target }) => {
+                    const value = (parentPath ? parentPath + FILE_SEPERATOR : '') + fileName
+                    setFileName(value)
+                    setParentPath('')
+                    setTimeout(() => {
+                      target.setSelectionRange(value.length, value.length)
+                      target.scrollLeft = Number.MAX_SAFE_INTEGER
+                    }, 0)
+                  }}
+                  placeholder="Name your file..."
```

Comment:
should this be added to the views.json?

Reply:
can be taken care of after pixel point team moves it to components-v2